packages/wallets/src/features/cashier/modules/WithdrawalCrypto/components/WithdrawalCryptoForm/components/WithdrawalCryptoAmountConverter/WithdrawalCryptoAmountConverter.tsx

```diff
@@ -0,0 +1,141 @@
+            return helperMessageMapper.decimalPlacesExceeded(FRACTIONAL_DIGITS_FIAT);
+
+        return undefined;
+    };
+
+    return (
+        <div className='wallets-withdrawal-crypto-amount-converter'>
+            <Field name='cryptoAmount' validate={validateCryptoInput}>
+                {({ field }: FieldProps<string>) => (
+                    <WalletTextField
+                        {...field}
+                        helperMessage={errors.cryptoAmount}
+                        label={`Amount (${activeWallet?.currency})`}
+                        onChange={e => {
+                            const value = parseFloat(e.target.value);
+                            const convertedValue =
+                                !Number.isNaN(value) && exchangeRate?.rates && activeWallet?.currency
+                                    ? (value / exchangeRate?.rates[activeWallet?.currency]).toFixed(
+                                          FRACTIONAL_DIGITS_FIAT
+                                      )
+                                    : '';
+                            setValues({
+                                ...values,
+                                cryptoAmount: e.target.value,
+                                fiatAmount: convertedValue,
+                            });
+                        }}
+                        onFocus={() => setIsCryptoInputActive(true)}
+                        showMessage
+                    />
+                )}
+            </Field>
+            <div
+                className={classNames('wallets-withdrawal-crypto-amount-converter__arrow', {
+                    'wallets-withdrawal-crypto-amount-converter__arrow--rtl': !isCryptoInputActive,
+                })}
+            >
+                <ArrowBold />
+            </div>
+            <Field name='fiatAmount' validate={validateFiatInput}>
+                {({ field }: FieldProps<string>) => (
+                    <WalletTextField
+                        {...field}
+                        helperMessage={errors.fiatAmount ?? 'Approximate value'}
+                        label='Amount (USD)'
+                        onChange={e => {
+                            const value = parseFloat(e.target.value);
+                            const convertedValue =
+                                !Number.isNaN(value) && exchangeRate?.rates && activeWallet?.currency
+                                    ? (value * exchangeRate?.rates[activeWallet?.currency]).toFixed(
+                                          FRACTIONAL_DIGITS_CRYPTO
+                                      )
+                                    : '';
+
+                            setValues({
+                                ...values,
+                                cryptoAmount: convertedValue,
+                                fiatAmount: e.target.value,
+                            });
```

Comment:
When the user enters the amount from the fiat-input the value of the `fiatAmount` is set directly as the value of the input and the value of `cryptoAmount` is set after converting the input value to crypto.